pyatlas/unit_tests/test_atlas.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+
+from pyatlas.atlas import Atlas
+from pyatlas import geometry
+from pyatlas.geometry import Rectangle
+
+
+class AtlasTest(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_lazy_loading(self):
+        atlas = Atlas("resources/test.atlas")
+        _touch_all_atlas_features(atlas)
+        self.assertEqual(atlas.number_of_points(), 5)
+        self.assertEqual(atlas.number_of_lines(), 2)
+        self.assertEqual(atlas.number_of_areas(), 2)
+        self.assertEqual(atlas.number_of_nodes(), 4)
+        self.assertEqual(atlas.number_of_edges(), 3)
+        self.assertEqual(atlas.number_of_relations(), 2)
+
+    def test_upfront_loading(self):
+        atlas = Atlas("resources/test.atlas", lazy_loading=False)
+        _touch_all_atlas_features(atlas)
+        self.assertEqual(atlas.number_of_points(), 5)
+        self.assertEqual(atlas.number_of_lines(), 2)
+        self.assertEqual(atlas.number_of_areas(), 2)
+        self.assertEqual(atlas.number_of_nodes(), 4)
+        self.assertEqual(atlas.number_of_edges(), 3)
+        self.assertEqual(atlas.number_of_relations(), 2)
+
+    def test_point_spatial_index(self):
+        atlas = Atlas("resources/test.atlas")
+
+        lower_left = geometry.location_with_degrees(37, -118.02)
+        upper_right = geometry.location_with_degrees(39, -118)
+
+        # NOTE point 1 does not show up in the results because it lies on the polygon border
+        test_results = atlas.points_within(Rectangle(lower_left, upper_right))
+        self.assertEquals({atlas.point(2), atlas.point(3)}, test_results)
+
+        test_results = atlas.points_within(
+            Rectangle(lower_left, upper_right), lambda p: p.get_identifier() % 2 != 0)
+        self.assertEquals({atlas.point(3)}, test_results)
+
+        test_results = atlas.points_at(geometry.location_with_degrees(38, -118))
+        self.assertEquals({atlas.point(1)}, test_results)
```

Comment:
Have these been tested and verified against the Java version?

Reply:
There is no automatic verification, but I have tested it manually against the Java version with the help of a script not included in this PR. Based on my tests the query results are consistent. But more tests are always welcome!